cpp/service.cc

```diff
@@ -0,0 +1,165 @@
+  const std::deque<std::string>& follow_users_values = store_->get(to_follow_user);
+  if (follow_users_values.size() == 0) {
+    return false;
+  }
+  const std::string this_user_key = kuserChirps_ + username;
+  const std::deque<std::string>& this_user_info = store_->get(this_user_key);
+  if (this_user_info.size() == 0) {
+    return false;
+  }
+  const std::string following_user_key = kuserFollowing_ + username;
+  store_->put(following_user_key, to_follow);
+  return true;
+}
+
+std::deque<chirp::Chirp> ServiceLayer::read(const std::string& chirp_id) {
+  std::deque<chirp::Chirp> read_chirps;
+
+  // Stores chirps to read
+  std::deque<std::string> chirp_list;
+  std::string given_chirp = chirp_id;
+  chirp_list.push_back(given_chirp);
+
+  while (chirp_list.size() > 0) {
+    const std::string curr_chirp_id = chirp_list.at(0);
+    chirp_list.pop_front();
+
+    if (curr_chirp_id.length() > 0) {
+      // Adds chirps to the deque of chirps in this thread
+      chirp::Chirp thisChirp;
+      const std::string this_chirp_key = kchirpValue_ + curr_chirp_id;
+      const std::deque<std::string>& this_chirps_values = store_->get(this_chirp_key);
+      thisChirp.ParseFromString(this_chirps_values.at(0));
+      read_chirps.push_back(thisChirp);
+
+      // Adds ids of thisChirp's replies to chirp_list to be read
+      const std::string this_chirp_reply_key = kchirpReplies_ + curr_chirp_id;
+      const std::deque<std::string>& this_chirp_replies = store_->get(this_chirp_reply_key);
+      for (const std::string reply : this_chirp_replies) {
+        chirp_list.push_back(reply);
+      }
+    }
+  }
+  std::sort(read_chirps.begin(), read_chirps.end(), [](const chirp::Chirp& chirp1, const chirp::Chirp& chirp2){
+    return chirp1.timestamp().useconds() < chirp2.timestamp().useconds();
+  });
+
+  return read_chirps;
+}
+
+std::deque<chirp::Chirp> ServiceLayer::monitor(const std::string& username, chirp::Timestamp start) {
+  std::deque<chirp::Chirp> found_chirps;
+  const std::string user_following_key = kuserFollowing_ + username;
+  const std::deque<std::string>& user_following = store_->get(user_following_key);
+  for (std::string username : user_following) {
+    if (username.length() > 0) {
+      const std::deque<std::string>& user_chirp_ids = store_->get(kuserChirps_ + username);
+      for (const std::string& id : user_chirp_ids) {
+        std::deque<std::string> this_chirps_values = store_->get(kchirpValue_ + id);
+        chirp::Chirp thisChirp;
+        if (this_chirps_values.size() > 0) {
+          thisChirp.ParseFromString(this_chirps_values.at(0));
+          int64_t myMicroSeconds = thisChirp.timestamp().useconds();
+          int64_t baslineSeconds = start.useconds();
+          std::string thisText = thisChirp.text();
+          if (thisChirp.timestamp().useconds() > start.useconds()) {
+            found_chirps.push_back(thisChirp);
+          }
+        }
+      }
+    }
+  }
+
+  return found_chirps;
```

Comment:
make all the const variables in this file start with a k

Reply:
As I mention above, it doesn't apply to all vars that have the `const` keyword, only those that are truly constants.